File: usedesk_sdk/src/main/java/ru/usedesk/sdk/external/ui/knowledgebase/pages/articlebody/ArticlesBodyViewModel.java
package ru.usedesk.sdk.external.ui.knowledgebase.pages.articlebody;

import android.support.annotation.NonNull;

import java.util.List;

import ru.usedesk.sdk.external.UsedeskKnowledgeBase;
import ru.usedesk.sdk.external.entity.knowledgebase.ArticleBody;
import ru.usedesk.sdk.external.ui.knowledgebase.common.DataViewModel;
import ru.usedesk.sdk.external.ui.knowledgebase.common.ViewModelFactory;

class ArticlesBodyViewModel extends DataViewModel<List<ArticleBody>> {

    private UsedeskKnowledgeBase usedeskKnowledgeBase;

    private ArticlesBodyViewModel(@NonNull UsedeskKnowledgeBase usedeskKnowledgeBase, @NonNull String searchQuery) {
        this.usedeskKnowledgeBase = usedeskKnowledgeBase;
        onSearchQueryUpdate(searchQuery);
    }

    public void onSearchQueryUpdate(@NonNull String searchQuery) {
        loadData(usedeskKnowledgeBase.getArticlesSingle(searchQuery));
    }

    static class Factory extends ViewModelFactory<ArticlesBodyViewModel> {
        private final UsedeskKnowledgeBase usedeskKnowledgeBase;
        private final String searchQuery;

        public Factory(@NonNull UsedeskKnowledgeBase usedeskKnowledgeBase, String searchQuery) {
            this.usedeskKnowledgeBase = usedeskKnowledgeBase;
            this.searchQuery = searchQuery;
        }

        @NonNull
        @Override
        protected ArticlesBodyViewModel create() {
            return new ArticlesBodyViewModel(usedeskKnowledgeBase, searchQuery);
        }

        @NonNull
        @Override
        protected Class<ArticlesBodyViewModel> getClassType() {
            return ArticlesBodyViewModel.class;
        }
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/pages/articlebody/ArticlesBodyViewModel.kt
package ru.usedesk.knowledgebase_gui.pages.articlebody

import ru.usedesk.knowledgebase_gui.common.DataViewModel
import ru.usedesk.knowledgebase_sdk.UsedeskKnowledgeBaseSdk
import ru.usedesk.knowledgebase_sdk.data.repository.entity.UsedeskArticleBodyOld

internal class ArticlesBodyViewModel : DataViewModel<List<UsedeskArticleBodyOld>>() {

    fun init(searchQuery: String) {
        onSearchQueryUpdate(searchQuery)
    }

    fun onSearchQueryUpdate(searchQuery: String) {
        loadData(UsedeskKnowledgeBaseSdk.getInstance().getArticlesRx(searchQuery))
    }
}